nixos/modules/services/audio/marytts.nix

```diff
@@ -0,0 +1,164 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.marytts;
+  format = pkgs.formats.javaProperties { };
+
+  maryBase = "/var/lib/marytts";
+in
+{
+  options.services.marytts = {
+    enable = lib.mkEnableOption "MaryTTS";
+
+    settings = lib.mkOption {
+      type = lib.types.submodule {
+        freeformType = format.type;
+      };
+      default = { };
+      description = ''
+        Settings for MaryTTS.
+
+        See the [default settings](https://github.com/marytts/marytts/blob/master/marytts-runtime/conf/marybase.config)
+        for a list of possible keys.
+      '';
+    };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 59125;
+      description = ''
+        Port to bind the MaryTTS server to.
+      '';
+    };
+
+    openFirewall = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      example = true;
+      description = ''
+        Whether to open the port in the firewall for MaryTTS.
+      '';
+    };
+
+    voices = lib.mkOption {
+      type = lib.types.listOf lib.types.path;
+      default = [ ];
+      example = lib.literalExpression ''
+        [
+          (pkgs.fetchzip {
+            url = "https://github.com/marytts/voice-bits1-hsmm/releases/download/v5.2/voice-bits1-hsmm-5.2.zip";
+            hash = "sha256-1nK+qZxjumMev7z5lgKr660NCKH5FDwvZ9sw/YYYeaA=";
+          })
+        ]
+      '';
+      description = ''
+        Paths to the JAR files that contain additional voices for MaryTTS.
+
+        Voices are automatically detected by MaryTTS, so there is no need to alter
+        your config to make use of new voices.
+      '';
+    };
+
+    userDictionaries = lib.mkOption {
+      type = lib.types.listOf lib.types.path;
+      default = [ ];
+      example = lib.literalExpression ''
+        [
+          (pkgs.writeTextFile {
+            name = "userdict-en_US";
+            destination = "/userdict-en_US.txt";
+            text = '''
+              amogus | V ' m O U g @ s
```

Comment:
Maybe a slightly less funny example for the docs.
What about Nix, NixOS or nixpkgs pronunciation? :)

(Or you can leave it as-is)

Could also do with a local path maybe? `/some/path/file.txt`

Reply:
I think Nixpkgs would be a good candidate cos it's fairly complicated as a word (?)